### ISSUE TITLE
[Web] Pub package google_sign_in crashes with v1.17.0

### ISSUE THREAD

**Issue by taronaeo** [flutter/flutter#54538]
## Steps to Reproduce

1. Upgrade Flutter from v1.15.17 to v1.17.0 on channel beta
2. Run `flutter run -d chrome` with an application that uses `google_sign_in`
3. Watch the errors spit out in the console

## Logs

```
D:\Projects\Flutter\RIVAlumni-old>flutter run -d chrome
Downloading Web SDK...                                              0.8s
Running "flutter pub get" in RIVAlumni-old...                       0.9s
Launching lib\main.dart on Chrome in debug mode...
Running "flutter pub get" in flutter_tools...                       0.6s
Syncing files to device Chrome...

Compiler message:
/D:/Flutter/.pub-cache/hosted/pub.dartlang.org/google_sign_in_web-0.8.3+2/lib/src/generated/gapiauth2.dart:26:3: Error: JS interop classes do not support non-external constructors.
Try annotating with `external`.
  GoogleAuth.fakeConstructor$();
  ^
/D:/Flutter/.pub-cache/hosted/pub.dartlang.org/google_sign_in_web-0.8.3+2/lib/src/generated/gapiauth2.dart:216:3: Error: JS interop classes do not support non-external constructors.
Try annotating with `external`.
  SigninOptionsBuilder.fakeConstructor$();
  ^
/D:/Flutter/.pub-cache/hosted/pub.dartlang.org/google_sign_in_web-0.8.3+2/lib/src/generated/gapi.dart:351:3: Error: JS interop classes do not support non-external constructors.
Try annotating with `external`.
  HttpRequestPromise.fakeConstructor$();
  ^
/D:/Flutter/.pub-cache/hosted/pub.dartlang.org/google_sign_in_web-0.8.3+2/lib/src/generated/gapi.dart:378:3: Error: JS interop classes do not support non-external constructors.
Try annotating with `external`.
  HttpRequest.fakeConstructor$() : super.fakeConstructor$();
  ^
/D:/Flutter/.pub-cache/hosted/pub.dartlang.org/google_sign_in_web-0.8.3+2/lib/src/generated/gapi.dart:402:3: Error: JS interop classes do not support non-external constructors.
Try annotating with `external`.
  HttpBatch.fakeConstructor$();
  ^
/D:/Flutter/.pub-cache/hosted/pub.dartlang.org/google_sign_in_web-0.8.3+2/lib/src/generated/gapi.dart:440:3: Error: JS interop classes do not support non-external constructors.
Try annotating with `external`.
  RpcRequest.fakeConstructor$();
  ^
Syncing files to device Chrome...                               13,651ms (!)
Failed to compile application.
```

```
D:\Projects\Flutter\RIVAlumni-old>flutter doctor -v
[√] Flutter (Channel beta, v1.17.0, on Microsoft Windows [Version 10.0.18363.720], locale en-SG)
    • Flutter version 1.17.0 at D:\Flutter
    • Framework revision d3ed9ec945 (5 days ago), 2020-04-06 14:07:34 -0700
    • Engine revision c9506cb8e9
    • Dart version 2.8.0 (build 2.8.0-dev.18.0 eea9717938)

[√] Android toolchain - develop for Android devices (Android SDK version 29.0.3)
    • Android SDK at D:\Android Studio SDK
    • Platform android-29, build-tools 29.0.3
    • ANDROID_SDK_ROOT = D:\Android Studio SDK
    • Java binary at: D:\Android Studio\jre\bin\java
    • Java version OpenJDK Runtime Environment (build 1.8.0_212-release-1586-b04)
    • All Android licenses accepted.

[√] Chrome - develop for the web
    • Chrome at C:\Program Files (x86)\Google\Chrome\Application\chrome.exe

[√] Android Studio (version 3.6)
    • Android Studio at D:\Android Studio
    • Flutter plugin version 44.0.2
    • Dart plugin version 192.7761
    • Java version OpenJDK Runtime Environment (build 1.8.0_212-release-1586-b04)

[√] Connected device (2 available)
    • Chrome     • chrome     • web-javascript • Google Chrome 81.0.4044.92
    • Web Server • web-server • web-javascript • Flutter Tools

• No issues found!
```

**Comment by eliudio** [comment]
I experience similar issue, on all channels (except stable).

Target dart2js failed: Exception:
/C:/dev/flutter/.pub-cache/hosted/pub.dartlang.org/google_sign_in_web-0.8.2/lib/src/generated/gapiauth2.dart:26:3:
Error: JS interop classes do not support non-external constructors.
  GoogleAuth.fakeConstructor$();
  ^
/C:/dev/flutter/.pub-cache/hosted/pub.dartlang.org/google_sign_in_web-0.8.2/lib/src/generated/gapiauth2.dart:216:3:
Error: JS interop classes do not support non-external constructors.
  SigninOptionsBuilder.fakeConstructor$();
  ^
/C:/dev/flutter/.pub-cache/hosted/pub.dartlang.org/google_sign_in_web-0.8.2/lib/src/generated/gapi.dart:351:3:
Error: JS interop classes do not support non-external constructors.
  HttpRequestPromise.fakeConstructor$();
  ^
/C:/dev/flutter/.pub-cache/hosted/pub.dartlang.org/google_sign_in_web-0.8.2/lib/src/generated/gapi.dart:378:3:
Error: JS interop classes do not support non-external constructors.
  HttpRequest.fakeConstructor$() : super.fakeConstructor$();
  ^
/C:/dev/flutter/.pub-cache/hosted/pub.dartlang.org/google_sign_in_web-0.8.2/lib/src/generated/gapi.dart:402:3:
Error: JS interop classes do not support non-external constructors.
  HttpBatch.fakeConstructor$();
  ^
/C:/dev/flutter/.pub-cache/hosted/pub.dartlang.org/google_sign_in_web-0.8.2/lib/src/generated/gapi.dart:440:3:
Error: JS interop classes do not support non-external constructors.
  RpcRequest.fakeConstructor$();
  ^
Error: Compilation failed.

C:\src\eliud\flutter_app>flutter doctor -v
[√] Flutter (Channel beta, v1.17.0, on Microsoft Windows [Version 10.0.18363.720], locale en-GB)
    • Flutter version 1.17.0 at C:\dev\flutter
    • Framework revision d3ed9ec945 (5 days ago), 2020-04-06 14:07:34 -0700
    • Engine revision c9506cb8e9
    • Dart version 2.8.0 (build 2.8.0-dev.18.0 eea9717938)

[!] Android toolchain - develop for Android devices (Android SDK version 29.0.2)
    • Android SDK at C:\Users\johan\AppData\Local\Android\sdk
    • Platform android-29, build-tools 29.0.2
    • Java binary at: C:\Program Files\Android\Android Studio\jre\bin\java
    • Java version OpenJDK Runtime Environment (build 1.8.0_212-release-1586-b04)
    X Android license status unknown.
      Try re-installing or updating your Android SDK Manager.
      See https://developer.android.com/studio/#downloads or visit visit https://flutter.dev/docs/get-started/install/windows#android-setup for detailed
      instructions.

[√] Chrome - develop for the web
    • Chrome at C:\Program Files (x86)\Google\Chrome\Application\chrome.exe

[√] Android Studio (version 3.6)
    • Android Studio at C:\Program Files\Android\Android Studio
    • Flutter plugin version 43.0.2
    • Dart plugin version 192.7761
    • Java version OpenJDK Runtime Environment (build 1.8.0_212-release-1586-b04)

[!] IntelliJ IDEA Community Edition (version 2019.2)
    • IntelliJ at C:\Program Files\JetBrains\IntelliJ IDEA Community Edition 2019.2.4
    X Flutter plugin not installed; this adds Flutter specific functionality.
    X Dart plugin not installed; this adds Dart specific functionality.
    • For information about installing plugins, see
      https://flutter.dev/intellij-setup/#installing-the-plugins

[√] VS Code (version 1.43.2)
    • VS Code at C:\Users\johan\AppData\Local\Programs\Microsoft VS Code
    • Flutter extension version 3.9.1

[√] Connected device (3 available)
    • Lenovo TB 8504F • 1234567890ABCDEF • android-arm64  • Android 7.1.1 (API 25)
    • Chrome          • chrome           • web-javascript • Google Chrome 81.0.4044.92
    • Web Server      • web-server       • web-javascript • Flutter Tools

! Doctor found issues in 2 categories.

**Comment by eliudio** [comment]
I experienced the same issue and found a solution that fixes this for me. Please find my answer here:

https://stackoverflow.com/questions/60911568/flutter-build-for-web-failed-to-compile-application/61167824#61167824

Basically you need to upgrade to google_sign_in_web 0.9.0. However, that conflicts with google_sign_in 4.4.1 which in itself conflicts with google_sign_in 0.8.2. So, clone google_sign_in, update dependency and use that temporarily, until they've upgraded google_sign_in

**Comment by taronaeo** [comment]
@eliudio Just to clarify, are you referring to the package `google_sign_in` or `google_sign_in_web`? Or are they both the same?

**Comment by eliudio** [comment]
@taronaeo No, they are not the same. I'm referring to the 2 different packages google_sign_in and google_sign_in_web.

If your project has a dependency on both, then follow the instructions given.

If your project only depends on google_sign_in_web, then just upgrade to 0.9.0

If your project only depends on google_sign_in, then follow these instructions as well, although you can just depend on a local copy of google_sign_in which depends on google_sign_in_web 0.9.0.

I hope this helps. It's just temporary, until the project flutter fixes it. I've created a pull request for the fix.

**Comment by taronaeo** [comment]
Yikes. Alright, I'll leave this issue open until the problem is fixed, thanks :)

**Comment by ditman** [comment]
The plugins repo is red right now. I'll get this fixed as soon as the CI in the repo passes again.

**Comment by ditman** [comment]
I'm currently addressing this.

**Comment by ditman** [comment]
@eliudio 's PR has been merged. I'll publish the package to pub.dev on Monday.

**Comment by taronaeo** [comment]
Looks like the issue has been resolved. Thanks @eliudio and @ditman for your quick responses and fixes. I feel safe to close this issue now 👍

**Comment by ditman** [comment]
I've just published [google_sign_in 4.4.3](https://pub.dev/packages/google_sign_in/versions/4.4.3).
